presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -155,7 +155,7 @@ public Page getOutput()
         // join probe page with the lookup source
         Counter lookupPositionsConsidered = new Counter();
         if (probe != null) {
-            while (true) {
+            while (!operatorContext.getDriverContext().getAbortSignal().isAborted()) {
```

Comment:
This operator makes sure to yield by the use of `lookupPositionsConsidered`, so this `while (true)` loop is limited. Why isn't that sufficient?